zhusuan/distributions/utils.py

```diff
@@ -282,22 +277,21 @@ def assert_positive_int32(value, name, need_scalar):
             raise TypeError(name + " must be integer")
         elif value <= 0:
             raise ValueError(name + " must be positive")
+    elif isinstance(value, (list, tuple, dict)):
+        raise TypeError(name + " should be a scalar (0-D Tensor).")
```

Comment:
remove this check, convert_to_tensor and assert_rank will raise in that case.

Reply:
I write the check because the doc says
> If `value` is the instance of built-in type, it will be checked directly.
Otherwise, it will be converted to a `tf.int32` tensor and checked.

Reply:
I realize it's not very easy to check if a type is a numeric type in python. And there is actually no need that we keep n_experiments a int. So it would be better that we leave all things to TF. We should remove all things in the if and do the convert directly.